WordPress/src/main/java/org/wordpress/android/GCMMessageService.java

```diff
@@ -122,20 +121,19 @@ private void handleDefaultPush(String from, @NonNull Bundle data) {
 
     private void trySaveNoteIfNotAlreadyInBucket(Bundle data) {
         if (SimperiumUtils.getNotesBucket() != null) {
-            Note note = null;
             String noteId = data.getString(PUSH_ARG_NOTE_ID, "");
             try {
-                note = SimperiumUtils.getNotesBucket().get(noteId);
+                SimperiumUtils.getNotesBucket().get(noteId);
                 // all good if we got here
             } catch (BucketObjectMissingException e) {
                 AppLog.e(T.NOTIFS, e);
                 SimperiumUtils.trackBucketObjectMissingWarning(e.getMessage(), noteId);
 
-                if (data != null && data.containsKey(PUSH_ARG_NOTE_FULL_DATA)) {
+                if (data.containsKey(PUSH_ARG_NOTE_FULL_DATA)) {
```

Comment:
if it was not because we know `trySaveNoteIfNotAlreadyInBucket` is only called with a non-null `data` param, I'd say not checking for null here makes this method risky to hit a NPE here. I would prefer to leave the null check as it was, just in case at some other point in time we decide to re-factor this and end up using this method elsewhere